recipes/pcapplusplus/cmake/conanfile.py

```diff
@@ -39,6 +40,10 @@ def validate(self):
         if self.settings.os not in ("FreeBSD", "Linux", "Macos", "Windows"):
             raise ConanInvalidConfiguration(f"{self.settings.os} is not supported")
 
+        compiler_version = Version(self.settings.compiler.version)
+        if self.settings.compiler == "gcc" and compiler_version < "5.1":
+            raise ConanInvalidConfiguration("PcapPlusPlus requires GCC >= 5.1")
```

Comment:
I added this because the build with GCC 5.0 fails:
https://c3i.jfrog.io/c3i/misc/logs/pr/25348/1-linux-gcc-legacy/pcapplusplus/24.09//67b8b30a48b5f4d20d090acd959766840b9443a5-build.txt
https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/25348/1-linux-gcc-legacy/pcapplusplus/24.09//summary.json

I'm not sure why initializing an `std::array` with an initialization list doesn't work, but [this thread](https://stackoverflow.com/a/8192275) suggests it might be related to a bug in old GCC versions

This is new code that was added in v24.09 I can add this validation only for v24.09, but I figured that GCC 5 is pretty old so this validation can apply to v23.09 as well. Please let me know what you think